### DISCUSSION TITLE
Export Subtitles (.ass and .srt) as QuickTime .mov with alpha

### DISCUSSION THREAD

**Discussion by Joelson-Forte** [SubtitleEdit/subtitleedit#8534] [Ideas]
Please, add an option to export your subtitles as QuickTime .mov video with with fully transparent background. The exported video can later be inserted as a separate video track in different NLE applications, including DaVinci Resolve, Final Cut Pro and Adobe Premiere, and thus be embedded in the video.

The only software I know that has this option is EZTiles. But it is quite expensive.
See in this link https://www.eztitles.com/Webhelp/EZTitles/export_subtitles_qtmov.htm

If you need it, I have the ffmpeg command line that does this.

**Comment by niksedk** [comment]
Could you supply the ffmpeg cmd line?

**Reply by Joelson-Forte** [reply]
Hi @niksedk 
The command is this. I use this command in a .bat file that must be in the same folder as the .ass or .srt file

ffmpeg -y -f lavfi -i "color=c=black@0.0:s=1920x1080:r=23.976:d=00\\:01\\:00,format=rgba,subtitles=C\\:\\File.ass:alpha=1" -c:v qtrle Video.mov

Sintax
s=Size (Resolution)
r=Rate (Frame Rate)
d=Duration (hh\\:mm\\:ss)
subtitles=Drive Letter: C,D,E,F,G,H...\\:\\File.ass
-c:v=Video Codec
Video=Name of Output Video

More finfo in this link: https://github.com/nilaoda/Blog/discussions/15 

Thanks for you job. Your software is amazing. He helps me a lot in my work

The exhaust bars in command are double. I don't know why GitHub leaves it that way in the message. Here's what they should look like.
![2024-06-16_16h07_46](https://github.com/SubtitleEdit/subtitleedit/assets/131564527/92d57b24-21b4-44d9-a953-96c668f81343)

**Reply by niksedk** [reply]
How is this?
https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.6/SubtitleEditBeta.zip

It's in the video menu:
![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/a756ffcc-3613-46d1-9d2c-7efb05a00838)

![image](https://github.com/SubtitleEdit/subtitleedit/assets/1261331/e242712a-663c-4e0a-827f-d3f853bc369b)

Uses something like `-y -f lavfi -i "color=c=black@0.0:s=1356x678:r=25:d=00\\:00\\:30,format=rgba,subtitles=f=fcbefccafbbcffa.ass:alpha=1" -c:v png "output.mp4"`

**Reply by Joelson-Forte** [reply]
I tried to do a little test but got only this
![2024-06-23_19h08_13](https://github.com/SubtitleEdit/subtitleedit/assets/131564527/1fbfb3f3-d895-410c-9f46-29ab7040737e)
![2024-06-24_17h17_43](https://github.com/SubtitleEdit/subtitleedit/assets/131564527/04ee0861-32c7-4f30-aa29-f5135877766e)
What makes it harder to implement is this command that uses escape slashes. I recently discovered that ffmpeg only recognizes the path to a folder if it is configured like this: \\\\\\\\Folder\\\\\\\\  The full command for an .ass or .srt file in a folder in the C:\\ directory would look like this:  ffmpeg -y -f lavfi -i "color=c=black@0.0:s=1920x1080:r=30:d=00\\\\:00\\\\:30,format=rgba,subtitles=C\\\\:\\\\\\\\Folder\\\\\\\\Test File.ass:alpha=1" -c:v qtrle Video.mov

**Reply by niksedk** [reply]
New beta upped: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.6/SubtitleEditBeta.zip
How does it work for you?

**Reply by Joelson-Forte** [reply]
Now works here... But it only processes the first 30 seconds of the subtitle. I think it's because the internal command that sets the duration is set that way. I made a little screen recording showing how it currently looks and how I will use it in my workflow.
[Subtitle Edit Screen Record.zip](https://github.com/user-attachments/files/15964673/Subtitle.Edit.Screen.Record.zip)

I have a small suggestion for this feature... do you think it's possible to add an option to generate only the selected area in the subtile?

![2](https://github.com/SubtitleEdit/subtitleedit/assets/131564527/b51f652d-e891-4f57-b0f8-f0a0f0bcd5f7)

**Reply by niksedk** [reply]
OK, added selection here: https://github.com/SubtitleEdit/subtitleedit/releases/download/4.0.6/SubtitleEditBeta.zip
How does it work?

**Reply by Joelson-Forte** [reply]
In the test I did, everything is working fine. The only thing I found strange was that when enabling the Opaque Box, all the opaque box settings (color and transparency) are applied to the font "Outline" option. Correcting this will be ready for use...

I want to take advantage and answer a question that I have had for a long time: Is there a code to set the font "Outline" thickness, color and transparency?

For example:
The code to choose the font is `<font face="Fonte Name">`
The code to choose the font color is `<font color="#ffe538">`
The code to choose the font size is `<font size="16px">`

**Reply by MbuguaDavid** [reply]
This is an awesome addition to Subtitle Edit. Thank you so much for this. 
I have  few requests:

1. Can we have a Preview window like we have in Burn in Feature
2. Batch Mode to Generate Transparent video with Subtitles
3. If I only need to use the Generate Transparent Video with Subtitles Feature only, it does not prompt to download FFmpeg. Can you make this available like in the Burn in Subtitles to video if one does not have FFmpeg installed?

I've not tested the latest version (yesterday's) but the other bug was generating a video that's 30 sec long not the length of the subtitle selected. I hope this was fixed. 


Thank you so much. :)
